### DISCUSSION TITLE
v1.62.0

### DISCUSSION THREAD

**Discussion by alextran1502** [immich-app/immich#2852] [Announcements]
# v1.62.0

## Highlights
Welcome to the `v1.62.0` release of Immich. There are many updates in this version that we hope you'll like. Some of the key highlights include:

* Thumbhash on the web.
* Easier to host the Machine Learning container on a different machine.

### Thumbhash on the web

The thumbnail now displays the blur version of the thumbnail image before the actual image is loaded. This mechanism replaces the transition from the gray box to the actual image to provide a better transition in the browsing experience. 

The video simulates a slow network loading scenario. 

https://github.com/immich-app/immich/assets/58121030/2c9e80e1-4baa-4f99-9d9e-392ff407be7b

### Host the `machine-learning` container on a different machine.

The server will now send the asset's content over HTTP to the machine learning container so that the container can be spun up on a different machine without the need to access the `UPLOAD_LOCATION` to run all of the machine learning tasks. 

This feature will provide a better experience when deploying the machine-learning container for users that host the Immich server on a low-end machine.

- - - -

And as always, bugs are fixed, and many other improvements also come with this release.

Please consider supporting the project.

## Support

<p align="center">
<img src="https://media.giphy.com/media/LStqgGESXW8XnuCv5y/giphy.gif" width="250" title="Loading ~4000 images/videos"> 
</p>

If you find the project helpful and helps you in some ways, you can support it via the following channel.

- Monthly donation via [GitHub Sponsors](https://github.com/sponsors/alextran1502)
- One-time donation via [GitHub Sponsors](https://github.com/sponsors/alextran1502?frequency=one-time&sponsor=alextran1502)
- [Librepay](https://liberapay.com/alex.tran1502/)
- [buymeacoffee](https://www.buymeacoffee.com/altran1502)
- Bitcoin: 1FvEp6P6NM8EZEkpGUFAN2LqJ1gxusNxZX

It is a great way to let me know that you want me to continue developing and working on this project for years to come.

🎉 Cheers! 🎉

### Mobile
* fix(mobile): Sort newest first for asset selection in album by @alextran1502 in https://github.com/immich-app/immich/pull/2833
* chore(mobile): Add debug build type suffix to the applicationId and version by @brighteyed in https://github.com/immich-app/immich/pull/2826

### Web
* fix(web): error when refreshing asset view in memory page by @alextran1502 in https://github.com/immich-app/immich/pull/2789
* chore(web): use svelte motion tweening for animation by @uhthomas in https://github.com/immich-app/immich/pull/2788
* chore(web): hide memory lane navigation when it's no longer possible to scroll by @uhthomas in https://github.com/immich-app/immich/pull/2791
* fix(web): not displaying assets in album after adding shared user by @alextran1502 in https://github.com/immich-app/immich/pull/2804
* feat(web): show number of assets in memory progress bar by @uhthomas in https://github.com/immich-app/immich/pull/2813
* fix(web): hide memory lane navigation properly on scaled resolutions by @uhthomas in https://github.com/immich-app/immich/pull/2819
* feat(web): add padding to memory asset navigation by @uhthomas in https://github.com/immich-app/immich/pull/2822

### Server
* feat(web,server): add thumbhash support by @lukehmcc in https://github.com/immich-app/immich/pull/2649
* refactor(server): album count by @jrasm91 in https://github.com/immich-app/immich/pull/2746
* refactor(server): access checks by @jrasm91 in https://github.com/immich-app/immich/pull/2776
* refactor(server): send job command by @jrasm91 in https://github.com/immich-app/immich/pull/2777
* refactor(server): rename get auth user decorator by @jrasm91 in https://github.com/immich-app/immich/pull/2778
* chore(server): organize imports by @jrasm91 in https://github.com/immich-app/immich/pull/2779
* fix(server): only show assets 'on-this-day' with thumbnails by @jrasm91 in https://github.com/immich-app/immich/pull/2851

### Machine learning
* feat(ml) backend takes image over HTTP by @zkhan93 in https://github.com/immich-app/immich/pull/2783

* docs: Update README.md by @phillibl in https://github.com/immich-app/immich/pull/2787
* docs: Update xmp-sidecars.md by @phillibl in https://github.com/immich-app/immich/pull/2785
* chore(deps): pin image digests by @uhthomas in https://github.com/immich-app/immich/pull/2754

## New Contributors
* @phillibl made their first contribution in https://github.com/immich-app/immich/pull/2787
* @renovate made their first contribution in https://github.com/immich-app/immich/pull/2739
* @dfcowell made their first contribution in https://github.com/immich-app/immich/pull/2825

**Full Changelog**: https://github.com/immich-app/immich/compare/v1.61.0...v1.62.0

<hr /><em>This discussion was created from the release <a href='https://github.com/immich-app/immich/releases/tag/'>v1.62.0</a>.</em>

**Comment by mineketchup** [comment]
Could someone give an example of how to configure machine learning on a different machine?
For example having a weak arm device as main server but running machine learning job on a PC on the same local network

**Reply by jrasm91** [reply]
Copy the compose file to your other machine and comment out/remove everything except the ML service. I think you may need to add the port section to expose 3003.

Start the container on the new machine

Comment out or remove the ML container from original compose on your main server.

Set the machine learning url env variable to the IP:port of your new machine and restart the stack.

**Reply by mineketchup** [reply]
Thanks! it was easy as i tought. I'll try it now

**Reply by mineketchup** [reply]
So I cofigured everything as you said (adding expose: 3003 on the machine-learning container) and I can see from "docker logs immich-microservices" that the server can't connects to the PC with `[Nest] 1  - 06/20/2023, 1:26:59 AM   ERROR [JobService] Unable to run job handler: Error: connect EHOSTUNREACH 192.168.1.27:3003
[Nest] 1  - 06/20/2023, 1:26:59 AM   ERROR [JobService] Error: connect EHOSTUNREACH 192.168.1.27:3003
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1494:16)
[Nest] 1  - 06/20/2023, 1:26:59 AM   ERROR [JobService] Object:
{
  "id": "1e6697e8-5ffe-4160-82d6-832a0c775df3"
}`

What else should I do?

**Reply by jrasm91** [reply]
Do you have a firewall or anything else that would prevent the docker container from being able to communicate with that machine?

**Reply by jrasm91** [reply]
You should be able to get a response from `http://192.168.1.27:3003/ping` on the new machine as well as on the original machine.

**Reply by mineketchup** [reply]
No, it's the first thing I controlled. On my PC I used "sudo ufw allow 3003" to open the port and on server "sudo ufw allow 3000 and 3001".

> You should be able to get a response from http://192.168.1.27:3003/ping on the new machine as well as on the original machine.

I'll try it now. Thanks for the help

**Reply by jrasm91** [reply]
In addition to the firewall, just want to make sure in the docker compose configuration you're actually exposing the port on the container itself to the host. This is what we do for the proxy and you need to do the same for the ML container, but it'll be 3003 on both sides.

![image](https://github.com/immich-app/immich/assets/4334196/c1f49339-4da5-48e3-9a1e-eb058837c3bd)

**Reply by mineketchup** [reply]
Thanks a lot, that did the work.
I initially used `expose: 3003` instead of ports and that was a mistake.

I had other errors, too (which was just caused by me because apparently I commented out immich-web too, in fact I was really confused when I saw that I had 2 orphan containers)

And I got an internet black out exactly when I was troubleshooting so it took me a while to respond and to verify that everything was working.

**Reply by fabiannagel** [reply]
I followed your comments and I'm able to call the ping endpoint on the ML container from my host (outside Docker though). Upon manually triggering the face recognition job, the logs in my hosts's `immich_microservices` container are getting flooded with these:

```
[Nest] 1  - 07/18/2023, 12:36:53 PM   ERROR [JobService] Unable to run job handler: Error: connect EADDRNOTAVAIL ::1:80 - Local (:::0)
[Nest] 1  - 07/18/2023, 12:36:53 PM   ERROR [JobService] Error: connect EADDRNOTAVAIL ::1:80 - Local (:::0)
    at internalConnect (node:net:1066:16)
    at defaultTriggerAsyncIdScope (node:internal/async_hooks:464:18)
    at node:net:1250:9
    at process.processTicksAndRejections (node:internal/process/task_queues:77:11)
```

Any ideas what I might have missed? I'm happy to provide a documentation PR afterwards.

**Reply by jrasm91** [reply]
Can you ping it by ip from inside the immich_microservices container? And what is your MACHINE_LEARNING_URL environment variable set to?

**Reply by fabiannagel** [reply]
Yeah I can also ping from within the microservices container:

```
foo@bar:~/immich-app $ sudo docker exec -it 440380f27189 bash
440380f27189:/usr/src/app# curl http://192.168.178.38:3003/ping
"pong"440380f27189:/usr/src/app#
```

This is my `.env` file:

```
IMMICH_MACHINE_LEARNING_URL=http://192.168.178.38:3003
IMMICH_MACHINE_LEARNING_URL=true
```

**Reply by jrasm91** [reply]
Well, you are setting (overridding) the `IMMICH_MACHINE_LEARNING_URL` to `true`. You can only have one value per environment variable :-) Just delete the second line and try again.

**Reply by fabiannagel** [reply]
Oh lord, I should take a break. Cheers. Works fine now. 🤓

**Comment by muety** [comment]
Just came across this app and only wanted to express how amazing it is to me! :tada: This is pretty much exactly what I've been looking for for ages.

Would love to know when you consider it "stable" enough to actually use it as an alternative to Google Photos?

**Reply by alextran1502** [reply]
Hello, we don't have an ETA on when it will be stable. I am using it as my primary driver with my wife at home. Make sure you have proper backup plan in place and you can enjoy Immich or any other application :)

**Comment by fabiannagel** [comment]
Thanks a lot for all the hard work! Immich is already amazing.

How robust is Immich when it comes to hosting the ML container on a different machine? Does it have to be up all the time or can it deal with a partially available ML container? I was thinking of using my desktop machine to do all the facial recognition stuff in the background while I'm working on other things.

**Reply by mineketchup** [reply]
So, using for several days it seems to be very consistent. Sometimes I power off the machine containing ml-service and the remaining works without any issue, so you shouldn't have any issue
